src/Distribution/Extra/Doctest.hs

```diff
@@ -435,3 +446,33 @@ testDeps :: ComponentLocalBuildInfo -> ComponentLocalBuildInfo
          -> [(InstalledPackageId, PackageId)]
 #endif
 testDeps xs ys = nub $ componentPackageDeps xs ++ componentPackageDeps ys
+
+amendGPD
+    :: String -- ^ doctests test-suite name
+    -> GenericPackageDescription
+    -> GenericPackageDescription
+#if !(MIN_VERSION_Cabal(2,0,0))
+amendGPD _ = id
+#else
+amendGPD testSuiteName gpd = gpd
+    { condTestSuites = map f (condTestSuites gpd)
+    }
+  where
+    f (name, condTree)
+        | name == fromString testSuiteName = (name, condTree')
+        | otherwise                        = (name, condTree)
+      where
+        -- I miss 'lens'
+        testSuite = condTreeData condTree
+        bi = testBuildInfo testSuite
+        om = otherModules bi
+        am = autogenModules bi
+
+        -- append
+        om' = fromString "Build_doctests" : om
+        am' = fromString "Build_doctests" : am
+
+        bi' = bi { otherModules = om', autogenModules = am' }
```

Comment:
I'm a bit surprised that you have to add `Build_doctests` to both `other-modules` and `autogen-modules`! Is this by design?

Reply:
Unfortunaly yes. There is a ticket (IIRC) to add `autogen-modules` to `other-modules` automatically, if they aren't `exposed-modules`. But it's won't be there even for `cabal-version: 2.2`.